actors/builtin/paych/paych_test.go

```diff
@@ -23,158 +25,707 @@ func TestPaymentChannelActor_Constructor(t *testing.T) {
+		builder := mock.NewBuilder(ctx, paychAddr).
+			WithCaller(callerAddr, builtin.InitActorCodeID).
+			WithActorType(payerAddr, builtin.AccountActorCodeID)
+		rt := builder.Build(t)
+		rt.ExpectValidateCallerType(builtin.InitActorCodeID)
+		rt.ExpectAbort(exitcode.ErrIllegalArgument, func() {
+			rt.Call(actor.Constructor, &ConstructorParams{To: paychAddr})
+		})
+	})
+}
+
+func TestPaymentChannelActor_CreateLaneSuccess(t *testing.T) {
+	ctx := context.Background()
+	rt, _, sv := requireCreateChannelWithLanes(t, ctx, 1)
+	var st State
+	rt.GetState(&st)
+	assert.Len(t, st.LaneStates, 1)
+	ls := st.LaneStates[0]
+	assert.Equal(t, sv.Amount, ls.Redeemed)
+	assert.Equal(t, sv.Nonce, ls.Nonce)
+	assert.Equal(t, sv.Lane, ls.ID)
 }
 
-func TestPaymentChannelActor_UpdateChannelState(t *testing.T) {
+func TestPaymentChannelActor_CreateLaneFailure(t *testing.T) {
 	ctx := context.Background()
 	actor := pcActorHarness{Actor{}, t}
 
-	newPaychAddr := tutil.NewIDAddr(t, 100)
-	callerAddr := tutil.NewIDAddr(t, 101)
+	initActorAddr := tutil.NewIDAddr(t, 100)
+	paychAddr := tutil.NewIDAddr(t, 101)
 	payerAddr := tutil.NewIDAddr(t, 102)
 
-	// set up trivial signature verifier and hasher
-	versig := func(sig crypto.Signature, signer addr.Address, plaintext []byte) bool { return true }
-	hasher := func(data []byte) [8]byte {
-		return [8]byte{}
+	sig := &crypto.Signature{Type: crypto.SigTypeBLS, Data: []byte("doesn't matter")}
+
+	testCases := []struct {
+		desc       string
+		targetCode cid.Cid
+
+		balance  int64
+		received int64
+		epoch    int64
+
+		tl    int64
+		lane  uint64
+		nonce uint64
+		amt   int64
+
+		secretPreimage []byte
+		sig            *crypto.Signature
+		verifySig      bool
+		expExitCode    exitcode.ExitCode
+	}{
+		{desc: "fails if balance too low", targetCode: builtin.AccountActorCodeID,
+			amt: 10, epoch: 1, tl: 1,
+			sig: sig, verifySig: true,
+			expExitCode: exitcode.ErrIllegalState},
+		{desc: "fails if new send balance is negative", targetCode: builtin.AccountActorCodeID,
+			amt: -1, epoch: 1, tl: 1,
+			sig: sig, verifySig: true,
+			expExitCode: exitcode.ErrIllegalState},
+		{desc: "fails if signature not valid", targetCode: builtin.AccountActorCodeID,
+			amt: 1, epoch: 1, tl: 1,
+			sig: nil, verifySig: true,
+			expExitCode: exitcode.ErrIllegalArgument},
+		{desc: "fails if too early for voucher", targetCode: builtin.AccountActorCodeID,
+			amt: 1, epoch: 1, tl: 10,
+			sig: sig, verifySig: true,
+			expExitCode: exitcode.ErrIllegalArgument},
+		{desc: "fails if signature not verified", targetCode: builtin.AccountActorCodeID,
+			amt: 1, epoch: 1, tl: 1, sig: sig, verifySig: false,
+			expExitCode: exitcode.ErrIllegalArgument},
+		{desc: "fails if SigningBytes fails", targetCode: builtin.AccountActorCodeID,
+			amt: 1, epoch: 1, tl: 1, sig: sig, verifySig: true,
+			secretPreimage: make([]byte, 2<<21),
+			expExitCode:    exitcode.ErrIllegalArgument},
```

Comment:
These are great, but without a success case it's hard to be confident that there isn't something that causes all these to fail that's not the thing actually being exercised (tho I have no doubt you verified in a debugger). Can you add a success case so it's clear that each of these other cases differs from that by only one factor?

Can possibly replace TestPaymentChannelActor_CreateLaneSuccess, tho not necessary.